### PR TITLE
Feature/ts

### PR DESCRIPTION
**Pull request by jaburcodes**

### CODE REVIEW

File: src/type/QueryType.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 //
```
Comment (gtkatakura) [inline src/type/QueryType.ts]: this line is unnecessary now

File: src/type/MutationType.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+//
```
Comment (gtkatakura) [inline src/type/MutationType.ts]: this line is unnecessary now, like any other files

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "graphql-dataloader-boilerplate-ts",
```
Comment (jgcmarins) [inline package.json]: do not remote `ts`

File: package.json [hunk 4]
```diff
@@ -27,25 +27,45 @@
     "mongoose": "^5.0.8",
     "subscriptions-transport-ws": "^0.9.6"
   },
+  "resolutions": {
+    "babel-core": "^7.0.0-bridge.0"
```
Comment (jgcmarins) [inline package.json]: what is this?

File: src/modules/user/rename.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Rename all *.txt to *.text
+for f in *.js; do 
+    mv -- "$f" "${f%.js}.ts"
+done
```
Comment (jgcmarins) [inline src/modules/user/rename.sh]: nice

File: src/rename.sh~ [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Rename all *.txt to *.text
+for f in *.txt; do 
+    mv -- "$f" "${f%.js}.ts"
+done
```
Comment (jgcmarins) [inline src/rename.sh~]: do not commit this file

File: package.json (hunk 4 above)
Comment (jaburcodes) [reply package.json]: @gtkatakura sent me this: http://artsy.github.io/blog/2017/11/27/Babel-7-and-TypeScript/

there its described that:
`it overrides child dependency versions with a fixed number set directly in package.json.` o/

File: index.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require('@babel/register')({
```
Comment (sibelius) [inline index.ts]: avoid babel register

use babel-node instead

**Comment by sibelius** [comment]
let's improve in next PRs

**Pull request merged**